### PR TITLE
46 new replies tabbed

### PR DESCRIPTION
**Pull request by GoreStarry**
* the tab of search 
* list of search articles
* modal for list article replies
* add overflow hidden for body when modal open
* List of search replies and search articles for redux articleDetail

### CODE REVIEW

File: components/ReplySearch/ReplySearch.js [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+
+    return (
+      <div>
+        <label htmlFor="replySeach">
+          搜尋相關回應：
+          <input id="replySeach" type="search" onKeyUp={this.handleSearch} />
+        </label>
+
+        {articles.size || replies.size
+          ? [
```
Comment (MrOrz) [inline components/ReplySearch/ReplySearch.js]: This array here introduces key errors:




Please use `<Fragment>` instead.

Although we did not upgrade babel to support `<>` and `</>`, we can still import `<Fragment>`:

```
import React, {Fragment} from React;
```

File: components/ReplySearch/ReplySearch.js [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+      </ul>
+    );
+  };
+
+  renderSearchReplyTab = () => {
+    const { articles, replies, onConnect } = this.props;
+    const { tab } = this.state;
+
+    const getArticleSimilarity = relatedArticleText =>
+      stringSimilarity.compareTwoStrings(this.search, relatedArticleText);
```
Comment (MrOrz) [inline components/ReplySearch/ReplySearch.js]: `this.search` is undefined. Isn't `search` in `this.state`?

Error screenshot:

File: ducks/articleDetail.js [hunk 3]
```diff
@@ -300,6 +410,37 @@ export default createReducer(
...
+          'node',
+          'replyConnections',
+          '0',
+          'article',
+        ]);
+        const version = reply.getIn(['node', 'versions']);
+        return Map()
+          .setIn(['id'], reply.getIn(['node', 'id']))
+          .setIn(['article'], article)
+          .setIn(['versions'], version);
```
Comment (MrOrz) [inline ducks/articleDetail.js]: Let's do this to simplify things up:
```
Map({
  id: reply.getIn(['node', 'id']),
  article: reply.getIn(['node', 'replyConnections', 0, 'article']),
  versions: reply.getIn(['node', 'versions']),
})
```

File: components/ReplySearch/ReplySearch.js (hunk 2 above)
Comment (GoreStarry) [reply components/ReplySearch/ReplySearch.js]: HAHA... big reconstruct mistake, THX!!

**Pull request merged**